test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperExecutionContextTest.cs

```diff
@@ -203,6 +203,54 @@ public void AddHtmlAttribute_MaintainsHTMLAttributes()
         }
 
         [Fact]
+        public void AddHtmlAttribute_MaintainsHTMLAttributes_Minimized()
+        {
+            // Arrange
+            var executionContext = new TagHelperExecutionContext("input", selfClosing: true);
+            var expectedAttributes = new TagHelperAttributeList
+            {
+                ["checked"] = new TagHelperAttribute { Name = "checked", Minimized = true },
+                ["visible"] = new TagHelperAttribute { Name = "visible", Minimized = true }
+            };
+
+            // Act
+            executionContext.AddHtmlAttribute("checked");
+            executionContext.AddHtmlAttribute("visible");
+
+            // Assert
+            Assert.Equal(
+                expectedAttributes,
+                executionContext.HTMLAttributes,
+                CaseSensitiveTagHelperAttributeComparer.Default);
+        }
+
+        [Fact]
+        public void AddHtmlAttribute_MaintainsHTMLAttributes_Mixed()
+        {
+            // Arrange
+            var executionContext = new TagHelperExecutionContext("input", selfClosing: true);
+            var expectedAttributes = new TagHelperAttributeList
+            {
+                { "class", "btn" },
+                { "foo", "bar" }
+            };
+            expectedAttributes.Add(new TagHelperAttribute { Name = "checked", Minimized = true });
+            expectedAttributes.Add(new TagHelperAttribute { Name = "visible", Minimized = true });
+
+            // Act
+            executionContext.AddHtmlAttribute("class", "btn");
+            executionContext.AddHtmlAttribute("foo", "bar");
+            executionContext.AddHtmlAttribute("checked");
+            executionContext.AddHtmlAttribute("visible");
```

Comment:
as I mentioned in the product code, ick.  not at all obvious what this line does.